mkosi

```diff
@@ -1744,6 +1769,44 @@ def find_kernel_file(workspace_root, pattern):
         warn('More than one kernel file found, will use {}', kernel_file)
     return kernel_file
 
+def install_grub(args, workspace, loopdev, grub):
+    if args.bios_partno is None:
+        return
+
+    nspawn_params = [
+        "--bind-ro=/dev",
+        "--property=DeviceAllow=" + loopdev,
+        "--property=DeviceAllow=" + partition(loopdev, args.root_partno),
+    ]
+
+    run_workspace_command(
+            args, workspace, "%s-install" % grub,
+            "--modules=ext2 part_gpt", "--target=i386-pc",
+            loopdev, nspawn_params=nspawn_params)
+
+#    run_workspace_command(args, workspace, "/bin/bash",
+#                          nspawn_params=nspawn_params)
+    run_workspace_command(
+            args, workspace, "%s-mkconfig" % grub,
+            "--output=/boot/%s/grub.cfg" % grub,
+            nspawn_params=nspawn_params)
+
+    cmdline = ("blkid", "--match-tag=PARTUUID", "--output=value",
```

Comment:
to me it sounds the wrong way round, to create the partition with a random uuid, and then query it off the image again. I think it would be nicer to do this the other way round: pick the uuid in mkosi before calling sfdisk, and then simply remember it and use it here...